crates/nu-path/src/pwd_per_drive.rs

```diff
@@ -0,0 +1,366 @@
+                    }
+                )
+            )
+        } else {
+            None
+        }
+    }
+
+    /// Expand a relative path using the PWD-per-drive, return PathBuf
+    /// of absolute path.
+    /// Return None if path is not valid or can't get drive letter.
+    pub fn expand_path(&mut self, path: &Path) -> Option<PathBuf> {
+        let path_str = path.to_str()?;
+        if let Some(drive_letter) = Self::extract_drive_letter(path) {
+            if let Some(pwd) = self.get_pwd(drive_letter) {
+                // Combine current PWD with the relative path
+                let mut base = PathBuf::from(Self::ensure_trailing_separator(&pwd));
+                base.push(path_str.split_at(2).1); // Skip the "C:" part of the relative path
+                return Some(base)
+            }
+        }
+        None // Invalid path or has no drive letter
+    }
+
+    /// Extract the drive letter from a path (e.g., `C:test` -> `C`)
+    fn extract_drive_letter(path: &Path) -> Option<char> {
+        Some(path.to_str()
+            .and_then(|s| s.chars().next())
+            .filter(|c| c.is_ascii_alphabetic())?
+        )
+    }
+
+    /// Ensure a path has a trailing `\`
+    fn ensure_trailing_separator(path: &str) -> String {
+        if !path.ends_with('\\') && !path.ends_with('/') {
+            format!(r"{}\", path)
+        } else {
+            path.to_string()
+        }
+    }
+}
+
+// GetFullPathW
+fn get_full_path_name_w(path_str: &str) -> Option<String> {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStringExt;
+    use std::os::windows::ffi::OsStrExt;
+    use winapi::um::fileapi::GetFullPathNameW;
+
+    const MAX_PATH : usize = 260;
+    let mut buffer: [u16; MAX_PATH] = [0; MAX_PATH];
+
+    unsafe {
+        // Convert input to wide string.
+        let wide_path: Vec<u16> = OsString::from(path_str).encode_wide().chain(Some(0)).collect();
+        let length = GetFullPathNameW(
+            wide_path.as_ptr(),
+            buffer.len() as u32,
+            buffer.as_mut_ptr(),
+            std::ptr::null_mut(),
+        );
+
+        if length > 0 && (length as usize) < MAX_PATH {
+            let path = OsString::from_wide(&buffer[..length as usize]);
+            if let Some(path_str) = path.to_str() {
+                let path_string = path_str.to_string();
+                {
+                    return Some(path_string);
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Global singleton instance of DrivePwdMap
+use std::sync::{Once, Mutex};
+
+static INIT: Once = Once::new();
+static mut DRIVE_PWD_MAP: Option<Mutex<Drive2PWDmap>> = None;
+
+/// Access the singleton instance
+fn get_drive_pwd_map() -> &'static Mutex<Drive2PWDmap> {
+    unsafe {
+        INIT.call_once(|| {
+            DRIVE_PWD_MAP = Some(Mutex::new(Drive2PWDmap::new()));
+        });
+        DRIVE_PWD_MAP.as_ref().unwrap()
+    }
+}
+
+/// Test for Drive2PWD map
+#[cfg(test)]
```

Comment:
We typically like to have our test in the crates/cratename/tests folder if possible. See other crates for guidance. I'm not sure nu-path has one yet.

Reply:
Unfortunately, nu-path only has 'src/', all tests are in-place in each module, I'll try to list 'doctest' at beginning as usage document, other tests to the end.

Reply:
We can live with this for now but we probably want to get this crate into shape like the other crates but we can save that for another PR.

Reply:
OK